### PR TITLE
Describe per-context limits in explainer

### PR DESCRIPTION
**Pull request by dmcardle**
~~This PR depends on #138. (GitHub doesn't understand dependent PRs, so this appears to include its parent's changes.)~~

This change is a little larger than it had to be, but I think it was time to add some structure to the section. WDYT, @alexmturner?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -380,21 +380,40 @@ One consideration in the short term is that these calls may have different
...
+- *Default limit:* The simplest strategy is to use one limit for all reports.
+  Our implementation currently enforces a limit of 20 contributions per report.
+
+- *Default limit per API:* The limit may be determined based on the calling API.
+  In particular, Protected Audience reports may benefit from a higher limit more
+  than Shared Storage reports. Our implementation plan is to set the limit at 20
+  contributions per report for Shared Storage and 100 contributions per report
+  for Protected Audience.
+
+- *Per-context limit:* Callers may request a different limit on each isolated
```
Comment (dmcardle) [inline README.md:405]: AKA "eager report sizing" and "per-caller sizing," which I think is a little ambiguous relative to the previous bullet.

File: README.md [hunk 2]
```diff
@@ -380,21 +380,42 @@ One consideration in the short term is that these calls may have different
...
+  of 20 contributions per report.
+
+- *Per-API limit:* The browser may select the limit based on the identity of the
+  calling API. In particular, Protected Audience reports may benefit from a
+  higher limit more than Shared Storage reports. Our implementation plan is to
+  set the limit at 20 contributions per report for Shared Storage and 100
+  contributions per report for Protected Audience.
+
+- *Per-context limit:* Callers may request a different limit on each isolated
+  context they create. However, callers that possess cross-site information
```
Comment (alexmturner) [inline README.md:408]: nit: maybe we can cast this as -- As this affects the payload size, this has to be specified from outside an isolated context. Consequently, ...

File: README.md [hunk 3]
```diff
@@ -380,21 +380,42 @@ One consideration in the short term is that these calls may have different
...
+  cannot use this mechanism. Consequently, Protected Audience buyers cannot set
+  per-context limits. The browser must clamp excessively large values to some
+  maximum value. Our implementation plan is to clamp the requested limit to a
+  maximum of 1000 contributions per report.
+
+- *Per-site limit:* A more complex design that enables sites to configure a
```
Comment (alexmturner) [inline README.md:414]: Maybe we can call this "global config"? (There's a chance we could make this config per origin)

**Review by alexmturner — COMMENTED** [review]
> This PR depends on #138. (GitHub doesn't understand dependent PRs, so this appears to include its parent's changes.)
> 
If your base branch is part of this repo, it is possible to mark the "base" branch of this PR as your other branch. Maybe I should work out how to give you the appropriate permissions...

> This change is a little larger than it had to be, but I think it was time to add some structure to the section. WDYT, @alexmturner?

Added a few comments, thanks!

File: README.md [hunk 4]
```diff
@@ -380,21 +380,42 @@ One consideration in the short term is that these calls may have different
...
 before truncation.
 
-If necessary, we could instead split the contributions back into multiple
-reports, each respecting the limit.
-
-Strawman limit: 20 contributions per report.
+Selecting the limit's value presents a tradeoff. Although larger reports have
+higher utility, they are also more expensive to process on the aggregation
+service. To accommodate use cases with diverse utility requirements and cost
+tolerances, we define several limit-selection strategies below.
```
[the remark this answers is not in the document]
Comment (dmcardle) [reply README.md:395]: Done

**Comment by dmcardle** [comment]
Hey @alexmturner, PTAL again :)

**Review by alexmturner — APPROVED** [review]

**Pull request merged**